test/withDataProviders.test.js

```diff
@@ -69,6 +69,23 @@ test('nested withDataProviders with eager prefetching - child component is rende
   expect(renderedMessage.textContent).toBe('1')
 })
 
+test('nested withDataProviders with eager prefetching - child component is rendered immediately, but ' +
+  'getData is called only once', async () => {
+  const getData = [getDataWithCount, {data: ''}, GET_DATA_DELAY]
+  const MessageContainer = messageContainer({needed: true, getData})
+  const ParentContainer = () => (<MessageContainer />)
+  const Container = compose(
+    withDataProviders(() => [messageProvider({needed: false, getData})]),
+  )(ParentContainer)
+  const {root} = renderApp(<Container />)
+
+  await safeDelay(GET_DATA_DELAY)
```

Comment:
I can't comment the line, but now that I see this I realized that there probably shouldn't be `safeDelay` on line 65 in previous test - it should test that after the data is eagerly pre-fetched, the needed child is displayed immediately.

Reply:
That's right, good catch, updated